### PR TITLE
Add mnemonic to launch-config shortcuts in context menu

### PR DESCRIPTION
**Pull request by mickaelistria**

### CODE REVIEW

File: org.eclipse.debug.ui/ui/org/eclipse/debug/ui/actions/ContextualLaunchAction.java [hunk 1]
```diff
@@ -242,7 +242,7 @@ protected void fillMenu(Menu menu) {
 						populateMenuItem(mode, ext, menu, null, accelerator++);
 					} else if (configurations.length > 0) {
 						for (ILaunchConfiguration configuration : configurations) {
-							populateMenuItem(mode, ext, menu, configuration, -1);
+							populateMenuItem(mode, ext, menu, configuration, accelerator++);
```
Comment (laeubi) [inline org.eclipse.debug.ui/ui/org/eclipse/debug/ui/actions/ContextualLaunchAction.java:245]: This was done by intention, there can only be 9 shortcuts and these should be reserved to the "real" shortcuts as one other whise easily runs out of shortcuts.

**Review by laeubi — CHANGES_REQUESTED** [review]

File: org.eclipse.debug.ui/ui/org/eclipse/debug/ui/actions/ContextualLaunchAction.java (hunk 1 above)
Comment (mickaelistria) [reply org.eclipse.debug.ui/ui/org/eclipse/debug/ui/actions/ContextualLaunchAction.java:245]: The expanded launch is often as important as the other shortcuts, so it also needs to be easily accessible, as easily as it used to be before we expanded the shortcut to launch.

Comment (laeubi) [reply org.eclipse.debug.ui/ui/org/eclipse/debug/ui/actions/ContextualLaunchAction.java:245]: But there is not **one** there can be many, for example m2e adds a lot of default shortcuts and then the available numbers are easily exhausted.

Comment (mickaelistria) [reply org.eclipse.debug.ui/ui/org/eclipse/debug/ui/actions/ContextualLaunchAction.java:245]: Sure there can be many, but still they can be as important as other shortcuts so should be also accessible with mneminics.

Comment (laeubi) [reply org.eclipse.debug.ui/ui/org/eclipse/debug/ui/actions/ContextualLaunchAction.java:245]: If you want that you can add a run config to the "favorites" and then you get a shortcut for a run config.

Comment (laeubi) [reply org.eclipse.debug.ui/ui/org/eclipse/debug/ui/actions/ContextualLaunchAction.java]: >  People who were used to hitting context-menu button, then R then 2 to run a JUnit test will miss it now.

See my comments below, but people that are used to do so might be even confused if hitting `R` then `2` suddenly executes *something else* and **not** a  JUnit test, that's why my initial intend was to not change the assignment but keep it intact with skipping additional (dynamic) run config.

**Comment by laeubi** [comment]
I dismissed my blocking review, because it seems you have a clear understanding and wanted to further improve and work on this feature but wanted to leave some thoughts:

1. In this menu we previously have only showed run **shortcuts**, and such a shortcut could actually do anything, most of the time it creates a **run configuartion**.
2. Initially I started with this to have an (invisible) **shortcut** that will be expanded into existing **run configurations** but this was now changed that shortcuts are always visible or replaced.
3. It was never the intention that this replaces other shortcuts, but it seems the there are many implementation that are now affected that way and therefore the "Shortcutproblem" and label confusions arise.

Therefore I think you probably want to enhance / extend the inital native assumptions implementation in the following way, this would be very beneficial for m2e (and my inital intention here):

1. If a shortcut return **exactly one run config**, keep the label of the shortcut and keep the shortcut
2. If there are more than one, use the label + launch name, but I think `Shortcutname (Run ConfigName)` would be much more suitable, maybe we can even have a (default implemented method) in `ILaunchShortcut2` that returns the name by passing the LaunchConfig to it so implementor can customize it, and only assign a shortcut to the first item
3. Have a (default implemented) methods in `ILaunchShortcut2` that allow to control the behavior, e.g. `isShortcutHidden()`, `isExpandLaunchConfigs()`, `isShowAccelerator()`...

This would most probably better match the "old" behavior people see today and much less surprising and be more flexible. Maybe you even want a `ILaunchShortcut3` interface for that purpose.

**Comment by SarikaSinha** [comment]
@mickaelistria 
Can you please provide a description to the PR ?
What is the aim and impact ?
Or do we have any issue where it is already discussed?

**Comment by mickaelistria** [comment]
@SarikaSinha I have added details, including link to initial issue, in the commit message.

**Comment by mickaelistria** [comment]
Although this might not be perfect, I believe this patch is a clear improvement over the current state. So I suggest we merge it and may come back to implement even more improvements later. when they appear necessary or even usefu

**Pull request merged**